### PR TITLE
Rename Subclass method

### PR DESCRIPTION
**Pull request by stuartwdouglas**
The current name has a high chance of conflicting, e.g. HttpFilter.destroy()

### CODE REVIEW

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
  */
 public interface Subclass {
 
-    default void destroy() {
+    default void destroy$subclass() {
```
Comment (mkouba) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java]: +1 for renaming but when I look at it, this method should not be part of the public API at all. We should remove this method from the `Subclass` interface and follow the naming convention we use for client proxies where an internal generated method is called `arc$delegate()`. So it should be `arc$destroy()` in this case.

**Review by mkouba — CHANGES_REQUESTED** [review]

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java [hunk 2]
```diff
@@ -7,8 +7,4 @@
  */
 public interface Subclass {
```
Comment (manovotn) [inline independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java]: With the method removed from here, this interface is now empty - does it make sense to keep it just as a marker interface?

Comment (mkouba) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java]: I believe it could be useful ;-)

Comment (manovotn) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java]: Hmm, I suppose you are right, after all, similar thing is in Weld for subclasses and for proxies...

**Review by manovotn — APPROVED** [review]

**Review by mkouba — APPROVED** [review]

File: independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java (hunk 2 above)
Comment (gsmet) [reply independent-projects/arc/runtime/src/main/java/io/quarkus/arc/Subclass.java]: Yes, please keep it, I use it :).

**Pull request merged**